### PR TITLE
fix: minify declarative JSON script content

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

Related to https://github.com/nuxt/nuxt/pull/35803

### ❓ Type of change

- [ ] 📖 Documentation
- [x] 🐞 Bug fix
- [ ] 👌 Enhancement
- [ ] ✨ New feature
- [ ] 🧹 Chore
- [ ] ⚠️ Breaking change

### 📚 Description

The runtime minifier skipped `speculationrules` and `importmap`, while the build transform skipped every declarative JSON script type. Both now strip insignificant whitespace while preserving numeric and escaped tokens. Compact JSON bypasses parsing, and invalid JSON stays unchanged.

## Summary by CodeRabbit

* **New Features**
  * JSON-based script content (`application/json`, `application/ld+json`, `speculationrules`, `importmap`) is now compacted automatically during builds and server-side rendering, including declarative `type` handling.

* **Bug Fixes**
  * JSON minification is now whitespace-stripping (no structural reformatting) and preserves dynamic content, invalid JSON, and encoded closing-tag-like sequences inside strings.

* **Documentation**
  * Updated minify plugin and build-time transform docs to reflect the expanded JSON script type support and behavior.

* **Tests / Chores**
  * Expanded minify/transform test coverage for JSON edge cases and added resolver aliases for `unhead/minify` in Vitest configs.

### CODE REVIEW

File: packages/bundler/src/unplugin/MinifyTransform.ts [hunk 1]
```diff
@@ -222,12 +232,14 @@ export const MinifyTransform = createUnplugin<MinifyTransformOptions, false>((op
...
+    let contentType: ContentType = tagType
     if (tagType === 'script') {
       const typeProp = objectNode.properties.find(
         (p: any) => p.type === 'Property' && p.key?.type === 'Identifier' && p.key.name === 'type',
       )
-      if (typeProp?.value?.type === 'Literal' && SKIP_JS_TYPES.has(typeProp.value.value))
+      if (typeProp?.value?.type === 'Literal' && JSON_TYPES.has(typeProp.value.value))
+        contentType = 'json'
+      else if (!doJS)
         return
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/bundler/src/unplugin/MinifyTransform.ts:239]: Fixed in b792bad4. Static quoted type keys now use the JSON path, with coverage using js: false.

File: packages/bundler/src/unplugin/MinifyTransform.ts [hunk 2]
```diff
@@ -26,6 +27,16 @@ interface PendingMinification {
 
 export type MinifyFn = (code: string) => Promise<string | null>
 
+const jsonMinifier: MinifyFn = async (code) => {
+  try {
+    return JSON.stringify(JSON.parse(code))
+  }
+  catch {
+    // Invalid declarative JSON stays untouched instead of being interpreted as JavaScript.
+    return code
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/bundler/src/unplugin/MinifyTransform.ts]: Fixed in b792bad4. minifyJSON now strips structural whitespace without reserializing tokens, so encoded < remains encoded. Added regression coverage for encoded closing tags and large numeric tokens.
